services/beehive-upload-server/user-service/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+// testing: docker run -d --name upload-server -p 8080:80  waggle/beehive-upload-server
+// curl localhost:8080/user
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type NodeObj struct {
+	ID string `json:"id"`
+}
+
+type APIResponse struct {
+	Data interface{} `json:"data"`
+}
+
+type APIErrorResponse struct {
+	Error string `json:"error"`
+}
+
+func GetUsers() (users []string, err error) {
+
+	file, err := os.Open("/etc/passwd")
+
+	if err != nil {
+		err = fmt.Errorf("error opening file: %s", err.Error())
+		return
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+
+	for {
+		var line string
+		line, err = reader.ReadString('\n')
+
+		// skip all line starting with #
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// get the username and description
+		lineSlice := strings.FieldsFunc(line, func(divide rune) bool {
+			return divide == ':' // we divide at colon
+		})
+
+		if len(lineSlice) > 0 {
+			user := lineSlice[0]
+			if !strings.HasPrefix(user, "node-") {
+				continue
+			}
+			users = append(users, lineSlice[0])
+		}
+
+		if err == io.EOF {
+			err = nil
+			break
+		}
+		if err != nil {
+			err = fmt.Errorf("error reading file: %s", err.Error())
+			return
+		}
+
+	}
+	if users == nil {
+		users = []string{}
+	}
+	return
+}
+
+func rootListener(w http.ResponseWriter, req *http.Request) {
+	fmt.Fprintf(w, "alive")
+}
+
+func userListener(w http.ResponseWriter, req *http.Request) {
+
+	fmt.Println("GET /user was called")
+	users, err := GetUsers()
+	if err != nil {
+		fmt.Printf("returned error: %s\n", err.Error())
+
+		ar := APIErrorResponse{}
+		ar.Error = err.Error()
+		response_json, _ := json.Marshal(ar)
+		http.Error(w, string(response_json), http.StatusInternalServerError)
+		return
+	}
+
+	ar := APIResponse{}
+	ar.Data = users
+	response_json, _ := json.Marshal(ar)
+	// err := Sync()
+
+	fmt.Fprint(w, string(response_json))
+
+}
+
+func CreateUser(username string) (err error) {
+	// adduser -D -g "" "$username"
+	// passwd -u "$username"
+	// chown -R "$username:$username" "/home/$username"
+	var out []byte
+	out, err = exec.Command("adduser", "-D", "-g", "\"\"", username).CombinedOutput()
```

Comment:
You may like the assign + check syntax for this, if you were worried about shadowing `out`. 

Something like:

```go
if out, err := exec.Command(args).CombinedOutput(); err != nil {
  // handle / return err
}
```